neural_compressor/torch/utils/utility.py

```diff
@@ -278,3 +280,61 @@ def get_model_device(model: torch.nn.Module):
     """
     for n, p in model.named_parameters():
         return p.data.device.type  # p.data.device == device(type='cpu')
+
+
+class ProcessorType(enum.Enum):
+    Client = "Client"
+    Server = "Server"
+
+
+def detect_processor_type_based_on_hw():
+    """Detects the processor type based on the hardware configuration.
+
+    Returns:
+        ProcessorType: The detected processor type (Server or Client).
+    """
+    # Detect the processor type based on below conditions:
```

Comment:
@xin3he @Kaihui-intel Here is the auto-detect logic. We can refine it if you have other suggestions.

Reply:
I'm not sure if this is correct. Also, maybe the number of cores is also a consideration.

Reply:
I enhanced the check a little bit and added more logs to inform the user of the detection result. If the detection result is wrong, the user can override it explicitly。